crates/proto/src/rr/domain/label.rs

```diff
@@ -75,6 +80,10 @@ impl Label {
     ///
     /// This will return an Error if the label is not an ascii string
     pub fn from_ascii(s: &str) -> ProtoResult<Self> {
+        if s.len() < 63 {
+            return Err(ProtoErrorKind::LabelBytesTooLong(s.len()).into());
+        }
+
```

Comment:
Took the liberty to add an early failure path. can remove it it's fine without it pass into `from_raw_bytes`

Reply:
All of these methods reduce to `from_raw_bytes`, which means that we have redundant checks. I think we could just rely on the check there, right? Are you concerned that someone may create a label without using from_raw_bytes?